### DISCUSSION TITLE
Can the priority of function signature keyword argument completions be configured?

### DISCUSSION THREAD

**Discussion by ben-davis** [microsoft/pyright#6535] [Q&A] [answered]
Often in large codebases, when typing a keyword argument of a function, many suggested auto-imports are shown before the `my_parameter=` is shown. Can this be configured? I often use keyword-only functions, so when I start typing, I'm typing the known parameter and so would love to have easy access to the completion.

**I'm using Neovim with `nvim-cmp`. Apologies if this is an issue with my editor, not pyright.**

For example, here I'm typing the `schema` parameter, but I get some auto-imports before that:
<img width="653" alt="Screenshot 2023-11-24 at 2 54 01 PM" src="https://github.com/microsoft/pyright/assets/2607068/897642a3-9a2d-4fbf-9e00-a4b8f5a8b2e0">

**Comment by erictraut** [comment, marked as answer]
There's currently no way to configure this. 

The logic for determining the order is developed and owned by the team responsible for Pylance, the language server that builds on top of pyright's type analysis capabilities. You could check the [pylance-release](https://github.com/microsoft/pylance-release) project to see if you can find a similar issue or discussion topic. If not, you could file an enhancement request there.

**Reply by ben-davis** [reply]
Thanks for your reply! I'm using Pyright within Neovim, so I'm assuming Pylance isn't involved and I'm just using Pyright's built in LSP. I'm not familiar with how the autocompletions work for an LSP; is it up to the client of the server to handle ordering the completions returned from the server? I assumed that as the logic of how the completions are created is heavily dependent on the inner workings of the LSP, it would be the server that dictates ordering.

In any case, would it be possible for a client to reorder giving priority to keyword arguments? Presumably a regex on the completion could be used, but I wonder if there's more metadata returned for a given completion?

Thanks for your help!

**Reply by erictraut** [reply]
Pyright includes some minimal language server functionality, which is how you're getting completion suggestions in Neovim. Pylance builds on pyright's functionality. The team responsible for Pylance also maintains the minimal language server functionality in the pyright code base, so if you want to suggest changes to that behavior, you'd need to convince the Pylance team to make those changes.

To answer your question, the language server is responsible for specifying the order of the completion suggestions. I suppose it's possible for a client to reorder the results that it receives from the server before presenting them, but the client would have little or no information about how to do so. That information is all internal to the language server.

**Reply by ben-davis** [reply]
Great, thanks very much for your help.
